app/questionnaire/path_finder.py

```diff
@@ -175,16 +175,16 @@ def _evaluate_routing_rules(
         routing_path_block_ids: list[str],
         when_rules_block_dependencies: list[str],
     ) -> int | None:
-        if when_rules_block_dependencies:
-            routing_path_block_ids = (
-                when_rules_block_dependencies + routing_path_block_ids
-            )
+        # Use `list` to create a shallow copy since routing_path_block_ids is mutated hence we don't to update its memory reference
```

Comment:
```suggestion
        # Use `list` to create a shallow copy since routing_path_block_ids is mutated hence we don't want to update its memory reference
```